contracts/vault/src/query.rs

```diff
@@ -17,11 +17,17 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<Binary> {
 
         QueryMsg::GetPendingWithdrawal { user } => {
             let pending = state::WITHDRAWAL_QUEUE
-                .load(deps.storage)?
-                .iter()
-                .filter(|req| req.user.to_string() == user)
-                .map(|req| req.amount)
+                .range(deps.storage, None, None, cosmwasm_std::Order::Ascending)
+                .filter_map(|item| {
+                    let (_, req) = item.ok()?;
+                    if req.user.to_string() == user {
+                        Some(req.amount)
+                    } else {
+                        None
+                    }
+                })
```

Comment:
This won't really scale for the same reasons. Instead it's better to add an additional map that will be updated at the same places as WITHDRAWAL_QUEUE. So something like:

```rust
pub const USER_WITHDRAWALS: Map<(&Addr, QueueId) , ()> = Map::new("user_withdrawals");
```

Then in this query here you can use [prefix_range](https://docs.rs/cw-storage-plus/latest/cw_storage_plus/struct.Map.html#method.prefix_range) with the `user` parameter which will iterate over key-value pairs of K: QueueId and V: (). So you can use the key here to load the entry from `WITHDRAWAL_QUEUE`.